### ISSUE TITLE
error while clicking on pie chart

### ISSUE THREAD

**Issue by sohampandya00** [imaNNeoFighT/fl_chart#146]
**Unhandled Exception: NoSuchMethodError: The method '[]' was called on null.
when i am tapping or long tapping some times. getting this error and after that i am not able to get touch event.** 

i solved it by changing in one file named "PieChartPainter". 

**sectionsAngle was null and we are getting value of it by index.** 

to temporary solve it. i made following changes.

1) adding try catch. 
2) checking section Angle is null or not and checking its length.

```
try {
      for (int i = 0; i < data.sections.length; i++) {
        final section = data.sections[i];
        if(sectionsAngle!=null)
        {
          if(sectionsAngle.length > i ) {
            double sectionAngle = sectionsAngle[i];

            tempAngle %= 360;
            sectionAngle %= 360;

            /// degree criteria
            final space = data.sectionsSpace / 2;
            final fromDegree = tempAngle + space;
            final toDegree = sectionAngle + tempAngle - space;
            final isInDegree = touchAngle >= fromDegree &&
                touchAngle <= toDegree;

            /// radius criteria
            final centerRadius = data.centerSpaceRadius;
            final sectionRadius = centerRadius + section.radius;
            final isInRadius = touchR > centerRadius && touchR <= sectionRadius;

            if (isInDegree && isInRadius) {
              foundSectionData = section;
              foundSectionDataPosition = i;
              break;
            }

            tempAngle += sectionAngle;
          }
        }
      }
    } on Exception catch(e)
    {
      print("caught");

    }
    catch(e)
    {
      print("caught");
    }
```
 
I dont know my way is right or not as i ddnt go through whole library code. but i am not getting this error now.  kindly guide.

**Comment by imaNNeoFighT** [comment]
Hi there,
I didn't understand the problem,
may I ask you to provide a reproducible code (include the main function)?
Thanks!

**Comment by msarkrish** [comment]
════════ Exception caught by gesture ═══════════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown while handling a gesture:
The method '[]' was called on null.
Receiver: null
Tried calling: [](0)

When the exception was thrown, this was the stack: 
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
#1      PieChartPainter._getTouchedDetails (package:fl_chart/src/chart/pie_chart/pie_chart_painter.dart:197:42)
#2      PieChartPainter.handleTouch (package:fl_chart/src/chart/pie_chart/pie_chart_painter.dart:170:12)
#3      PieChartState.build.<anonymous closure> (package:fl_chart/src/chart/pie_chart/pie_chart.dart:56:28)
#4      LongPressGestureRecognizer._checkLongPressEnd.<anonymous closure> (package:flutter/src/gestures/long_press.dart:333:52)
...
Handler: "onLongPressEnd"
Recognizer: LongPressGestureRecognizer#7f88e
  debugOwner: GestureDetector
  state: possible
════════════════════════════════════════════════════════════════════════════════════════════════════

i'm getting this error when long press the chart with given sample code.

**Comment by imaNNeoFighT** [comment]
Provide me a reproducer code.

**Comment by imaNNeoFighT** [comment]
It will be closed by tomorrow if you don't provide it.
Thanks!

**Comment by msarkrish** [comment]
I will provide code Today.

**Comment by imaNNeoFighT** [comment]
Thank you, I found the problem.
It will be fixed in the next version,
stay tuned!

**Comment by msarkrish** [comment]
Waiting for the next version brother.

**Comment by imaNNeoFighT** [comment]
Thanks for waiting,
the next version will be 0.7.0
stay tuned! 
Thanks!

**Comment by imaNNeoFighT** [comment]
Fixed in [0.7.0](https://github.com/imaNNeoFighT/fl_chart/releases/tag/0.7.0)
